infra/testApi.mjs

```diff
@@ -236,6 +181,113 @@ checkLocalErrorNode(localError7.tree, [0, 1],
+    return json;
+  }
+}
+
+// MathJS endpoint
+const test_json = await callTranslate(FPCoreFormula, 'mathjs');
+const expected_value = "sqrt(x + 1.0) - sqrt(x)";
+assert.equal(test_json.mathjs, expected_value);
+
+// Translate endpoint
+const expectedExpressions = {
+  "python": 'def expr(x):\n\treturn math.sqrt((x + 1.0)) - math.sqrt(x)\n',
+  "c": 'double expr(double x) {\n\treturn sqrt((x + 1.0)) - sqrt(x);\n}\n',
+  "fortran": 'real(8) function expr(x)\nuse fmin_fmax_functions\n    real(8), intent (in) :: x\n    expr = sqrt((x + 1.0d0)) - sqrt(x)\nend function\n',
+  "java": 'public static double expr(double x) {\n\treturn Math.sqrt((x + 1.0)) - Math.sqrt(x);\n}\n',
+  "julia": 'function expr(x)\n\treturn Float64(sqrt(Float64(x + 1.0)) - sqrt(x))\nend\n',
+  "matlab": 'function tmp = expr(x)\n\ttmp = sqrt((x + 1.0)) - sqrt(x);\nend\n',
+  "wls": 'expr[x_] := N[(N[Sqrt[N[(x + 1), $MachinePrecision]], $MachinePrecision] - N[Sqrt[x], $MachinePrecision]), $MachinePrecision]\n', // Wolfram 
+  "tex": '\\mathsf{expr}\\left(x\\right) = \\sqrt{x + 1} - \\sqrt{x}\n',
+  "js": 'function expr(x) {\n\treturn Math.sqrt((x + 1.0)) - Math.sqrt(x);\n}\n'
+}
+
+for (const e in expectedExpressions) {
+  const test_json = await callTranslate(FPCoreFormula, e);
+  assert.equal(test_json['language'], e);
+  assert.equal(test_json['result'], expectedExpressions[e]);
+}
+
+// MARK: Herbie Demo
+// Endpoints associated only with Herbie Demo
+
+// improve endpoint
+const improveResponse = await fetch(makeEndpoint(`/improve?formula=${encodeURIComponent(FPCoreFormula2)}`), { method: 'GET' })
+assert.equal(improveResponse.status, 200);
+let redirect = improveResponse.url.split("/");
+const job_iD = redirect[3].split(".")[0];
+// MARK: Timeline
+const timelineRSP = await fetchAndCheckRSPHeaders(`/timeline/${job_iD}`, { method: 'GET' }, CHECK_CORS);
+assert.equal(timelineRSP.status, 201);
+// const timeline = await timelineRSP.json()
+// assert.equal(timeline.length > 0, true)
+
+// improve-start endpoint
+const URIencodedBody = "formula=" + encodeURIComponent(FPCoreFormula)
+const startResponse = await fetch(makeEndpoint("/api/start/improve"), {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/x-www-form-urlencoded',
+  },
+  body: URIencodedBody
+})
+const testResult = (startResponse.status == 201 || startResponse.status == 202)
+assert.equal(testResult, true)
+const improveResultPath = startResponse.headers.get("location")
+let counter = 0
+let cap = 100
+// Check status endpoint
+let checkStatus = await fetch(makeEndpoint(improveResultPath), { method: 'GET' })
+/*
+This is testing if the /api/start/improve test at the beginning has been completed. The cap and counter is a sort of timeout for the test. Ends up being 10 seconds max.
+*/
+while (checkStatus.status != 201 && counter < cap) {
+  counter += 1
+  checkStatus = await fetch(makeEndpoint(improveResultPath), { method: 'GET' })
+  await new Promise(r => setTimeout(r, 100)); // ms
+}
```

Comment:
I guess it's fine but isn't this a for loop with an early exit? Also I notice that even when the status is successful you still wait for 100ms before exiting.